packages/scene-composer/src/components/panels/scene-components/AnimationComponentEditor.tsx

```diff
@@ -0,0 +1,70 @@
+import React, { useCallback, useContext } from 'react';
+import { FormField, Multiselect, SpaceBetween } from '@awsui/components-react';
+import { useIntl } from 'react-intl';
+
+import { IComponentEditorProps } from '../ComponentEditor';
+import { IAnimationComponentInternal, ISceneComponentInternal, useStore } from '../../../store';
+import { sceneComposerIdContext } from '../../../common/sceneComposerIdContext';
+import { OptionDefinition } from '@awsui/components-react/internal/components/option/interfaces';
+import { getClip } from '../../three-fiber/AnimationComponent/AnimationComponent';
+
+export type AnimationComponentEditorProps = IComponentEditorProps;
+
+export const AnimationComponentEditor: React.FC<AnimationComponentEditorProps> = ({
+  node,
+  component,
+}: AnimationComponentEditorProps) => {
+  const sceneComposerId = useContext(sceneComposerIdContext);
+  const updateComponentInternal = useStore(sceneComposerId)((state) => state.updateComponentInternal);
+  const AnimationComponent = component as IAnimationComponentInternal;
+  const animationList = getClip(AnimationComponent.uri);
+  const intl = useIntl();
+
+  
+  var selectedOptions =  AnimationComponent.currentAnimations?.map(animation => ({ label: animation, value: animation})) as OptionDefinition[]
+  const animationOptions = animationList?.map(animation => ({ label: animation.name, value: animation.name}))
+    
+  const onUpdateCallback = useCallback(
+    (componentPartial: any, replace?: boolean) => {
+      const componentPartialWithRef: ISceneComponentInternal = { ref: component.ref, ...componentPartial };
+      updateComponentInternal(node.ref, componentPartialWithRef);
+    },
+    [node.ref, component.ref],
+  );
+
+function updateAnimations(detail){
```

Comment:
fix indentation

Reply:
Turn into a useCallback to reduce rerender churn

Reply:
Linter off?